include/deal.II/lac/vector_memory.templates.h

```diff
@@ -32,7 +33,21 @@ template <typename VectorType>
 typename GrowingVectorMemory<VectorType>::Pool &
 GrowingVectorMemory<VectorType>::get_pool()
 {
-  static GrowingVectorMemory<VectorType>::Pool pool;
+  // Kokkos needs to be initialized before constructing the static Pool for
+  // vector types that use Kokkos.
+  // If Kokkos is initialized by deal.II, this make sure that it is finalized
+  // after the Pool has been destroyed.
+  // If Kokkos is not initialized by deal.II, we assume that Kokkos is not
+  // finalized past program end together with static variables and we need to
+  // make sure to empty the Pool when finalizing Kokkos so that the destruction
+  // of the Pool doesn't call Kokkos functions.
```

Comment:
I hate `GrowingVectorMemory` so much...

Reply:
I agree, that class is a bit weird. I'd rather prefer to explicitly manage pools at the places they make sense. (I've written my own `VectorMemory` classes several times to make them less intrusive than the `GrowingVectorMemory`.